File: src/main/java/itmo/is/project/model/trade/Trade.java
package itmo.is.project.model.trade;

import itmo.is.project.model.user.User;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "trade")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Trade {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

//    @NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

//    @NotNull
    @CreationTimestamp
    @Column(name = "time", nullable = false)
    private LocalDateTime time;

    @OneToMany(mappedBy = "trade", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<TradeItem> items;
}

File: src/main/kotlin/itmo/isproject/model/trade/Trade.kt
package itmo.isproject.model.trade

import itmo.isproject.model.IntIdEntity
import itmo.isproject.model.user.User
import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime
import java.util.*

@Entity
@Table(name = "trade")
class Trade(

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    var user: User? = null,

    @CreationTimestamp
    @Column(name = "time", nullable = false)
    var time: LocalDateTime? = null,

    @OneToMany(mappedBy = "trade", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
    var items: MutableList<TradeItem> = ArrayList()

) : IntIdEntity()
